Source/WebKit/UIProcess/Network/NetworkProcessProxy.cpp

```diff
@@ -1403,9 +1408,16 @@ void NetworkProcessProxy::sendPrepareToSuspend(IsSuspensionImminent isSuspension
 {
     if (!s_suspensionAllowedForTesting)
         return completionHandler();
-    
+
+    auto suspendCompletionHandler = [this, weakThis = WeakPtr { *this }, completionHandler = WTFMove(completionHandler)](bool needsLockedFileAssertion) mutable {
+        if (needsLockedFileAssertion && weakThis)
+            m_holdingLockedFileAssertion = ProcessAssertion::create(processID(), "Network Process is holding locked files"_s, ProcessAssertionType::FinishTaskInterruptable, ProcessAssertion::Mode::Sync);
```

Comment:
It looks like I would also need to task a FinishTaskInterruptable assertion on the UIProcess as well here, because otherwise the UIProcess could suspend before it handles the `NetworkProcessProxy::releaseLockedFileAssertion` IPC.

Reply:
Seems like I'm also supposed to use the version of `ProcessAssertion::create` that takes an `ExtensionProcess` rather than a pid.